src/states/main_game.rs

```diff
@@ -31,6 +31,12 @@ impl Default for MainGameState {
     fn default() -> Self {
         MainGameState {
             dispatcher: DispatcherBuilder::new()
+                .with(perception::SpatialGridSystem, "spatial_grid", &[])
```

Comment:
Maybe this should go after the movement system, otherwise, it will use the old position

Reply:
I spent some time studying the problem to see what possible problems we may  have, but gave up on it as I'm a little bit tired.
Anyway, I agree it's better to leave any AI related system to after physics and movement systems.
I don't have an argument for it besides that it sounds more intuitive, because we make decisions after movement.

Still, we would need a spatial grid resource that stores the old positions as well, because this will be really useful for more efficient collision detection.

Reply:
I was thinking that the perception systems should come first because they determine how creatures are going to move on that time step. 
To me, it's more intuitive to imagine the movement as being the last step, once all influences have been computed. Of course, there would be collision detection and correction after. 
But I would imagine the logic that influences movement to come before the system that actually moves the entities.

Reply:
Yes you are right that makes more sense.